experimental/dds/tree2/src/core/tree/anchorSet.ts

```diff
@@ -470,79 +440,86 @@ export class AnchorSet implements ISubscribable<AnchorSetRootEvents> {
 				index++;
 			}
 			// Sever the parent -> child connections
-			toMove = srcChildren.splice(numberBeforeMove, numberToMove);
+			nodes = srcChildren.splice(numberBeforeMove, numberToMove);
 			if (srcChildren.length === 0) {
 				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-				srcParent!.afterEmptyField(srcStart!.parentField);
+				srcParent!.afterEmptyField(src.parentField);
 			}
-		} else {
-			toMove = [];
 		}
 
-		if (dst === undefined) {
-			// Change is a delete.
-			// Moved items have already been un-parented, so just mark them as deleted.
-			this.deepDelete(toMove);
-			return;
-		}
+		return nodes;
+	}
 
-		// Get dst (and set parent for moved items)
-		let dstPath: PathNode | undefined;
-		if (toMove.length > 0) {
-			// There are anchors which are getting moved,
-			// therefor the destination needs to be created if it does not yet exist.
+	/**
+	 * Attach nodes to a parent.
+	 * @param count - number of siblings that are attached in the original tree.
+	 * @param dst - where the siblings are attached to.
+	 * @param offset - offset of the siblings.
+	 * @param nodes - siblings to attach.
+	 *
+	 * TODO: tests
+	 */
+	private attachNodes(count: number, dst: UpPath, offset: number, nodes: PathNode[]): void {
```

Comment:
This signature is a little confusing. Might be better for detach to return something that includes the count offset and nodes grouped into a single object so you can better document the fields and pass them around together.